reed_solomon.go

```diff
@@ -38,26 +37,8 @@ type reedSolomonSplitter struct {
 // a size splitting the shards and returns a ReedSolomonSplitter.
 func NewReedSolomonSplitter(r io.Reader, numData, numParity, size uint64) (
 	*reedSolomonSplitter, error) {
-	var fileSize int64
-	var err error
-	fi, ok := r.(files.FileInfo)
-	if ok {
-		fileSize, err = fi.Size()
-	}
-	// If not a FileInfo object, or fails to fetch a size, try reading
-	// the whole stream in order to obtain size (this is common for testing).
-	if !ok || err != nil {
-		// Not a file object, but we need to know the full size before
-		// being streamed for reed-solomon encoding.
-		// Copy it to a buffer as a last resort.
-		b, err := ioutil.ReadAll(r)
-		if err != nil {
-			return nil, err
-		}
-		fileSize = int64(len(b))
-		// Re-pack reader
-		r = bytes.NewReader(b)
-	}
+	bufr := bufio.NewReader(r)
```

Comment:
This gets around by doing copying for all scenarios, which is slow for CLI add. We should only need to handle the http add case.

Reply:
OK this may be fine... did all unit tests pass in this repo?